external-crates/move/crates/move-analyzer/src/symbols.rs

```diff
@@ -2750,20 +3011,50 @@ pub fn on_references_request(context: &Context, request: &Request, symbols: &Sym
                             });
                         }
                     }
-                    if locs.is_empty() {
-                        Some(serde_json::to_value(Option::<lsp_types::Location>::None).unwrap())
-                    } else {
-                        Some(serde_json::to_value(locs).unwrap())
-                    }
-                }
-                None => Some(serde_json::to_value(Option::<lsp_types::Location>::None).unwrap()),
-            }
+                    locs
+                })
+                .map(|locs| serde_json::to_value(locs).unwrap())
         },
     );
 }
 
+/// Helper function that take a DefInfo, checks if it represents
+/// a enum arm variable defintion, and if need be converts it
+/// to the one that represents an enum guard variable (which
+/// has immutable reference type regarldes of arm variable definition
+/// type).
+pub fn maybe_convert_for_guard(
+    def_info: &DefInfo,
+    use_fpath: &Path,
+    position: &Position,
+    symbols: &Symbols,
+) -> Option<DefInfo> {
+    let DefInfo::Local(name, ty, is_let, is_mut, arm_binder) = def_info else {
+        return None;
+    };
+    if !arm_binder {
+        return None;
+    }
```

Comment:
Could we just store the guard's `Option<Loc>` in this field?

Reply:
Yes. I was not sure it was necessary but after our recent conv I now understand why it's needed